### PR TITLE
perf(utils): extract lodash.merge

### PR DESCRIPTION
**Pull request by Duncan-zjp**
perf(utils): extract lodash.merge

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -23,4 +23,5 @@ pnpm-debug.log*
 *.sw?
 
 # Coverage directory used by tools like istanbul
-coverage
```
Comment (John60676) [inline .gitignore:26]: 这个配置不要改

File: src/core/utils/lodash/merge.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    ) {
+      if (isObject(oriArr[i]) && isObject(target[i])) return;
+      oriArr[i] = target[i];
+    }
+    isPlainObject(target[i]) && mergeObj(oriArr[i], target[i]);
+    isPlainArray(target[i]) && mergeArr(oriArr[i], target[i]);
+  }
+};
+
+const mergeObj = (oriObj: object, target: object) => {
```
Comment (John60676) [inline src/core/utils/lodash/merge.ts:18]: 类型不要用 `object` 换成 `Record<string, any>`

**Review by John60676 — CHANGES_REQUESTED** [review]

File: src/core/utils/index.ts [hunk 3]
```diff
@@ -8,10 +8,15 @@ export const isString = (val: unknown): val is string =>
   toTypeString(val) === '[object String]';
 export const isPlainObject = (val: unknown): val is Record<string, any> =>
   toTypeString(val) === '[object Object]';
+export const isPlainArray = (val: unknown): val is Record<string, any> =>
```
Comment (John60676) [inline src/core/utils/index.ts:11]: `Array.isArray`

**Review by John60676 — CHANGES_REQUESTED** [review]

File: src/core/utils/lodash/merge.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    isPlainObject(target[key]) && mergeObj(oriObj[key], target[key]);
+    isPlainArray(target[key]) && mergeArr(oriObj[key], target[key]);
+  }
+};
+
+function merge(origin: object, ...other: object[]) {
+  const result = Object.assign({}, origin);
+  if (!other.length) return result;
+
+  for (let i = 0; i < other.length; i++) {
```
Comment (benny-eu) [inline src/core/utils/lodash/merge.ts:37]: for...of

**Review by benny-eu — CHANGES_REQUESTED** [review]

**Review by John60676 — APPROVED** [review]

**Pull request merged**